### DISCUSSION TITLE
Chamada de Arquivo no Linux

### DISCUSSION THREAD

**Discussion by ernaneSNcarvalho** [balta-io/2801#4] [Perguntas e Respostas] [answered]
Comecei a usar o linux MINT e com esse código, nao consigo chamar meu arquivo txt no diretorio indicado. Como fazer ?

namespace` Archive1
{
    class Program
    {
        static void Main(string[] args)
        {
            string sourcePath = @"/Home/Documentos/Teste.txt";
            string targetPath = @"/Home/Documentos/Teste2.txt";
            
            try
            {
                FileInfo fileInfo = new FileInfo(sourcePath);
                fileInfo.CopyTo(targetPath);
            }
            catch(IOException e)
            {
                Console.WriteLine("An error occurred");
                Console.WriteLine(e.Message);
            }
        }
    }
}

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @ernaneSNcarvalho, como vai?
Consegue postar o erro que está recebendo?

**Reply by ernaneSNcarvalho** [reply]
An error occurred
Could not find a part of the path '/Home/Documentos/Teste.txt'.

**Reply by andrebaltieri** [reply]
Pelo que pude entender seu caminho está errado... Você consegue navegar e gravar arquivos neste caminho (`/Home/Downloads`)?
Estas pastas contém caracteres maiúsculos mesmo? O Linux é case sensitive, então `home` é diferente de `Home`....

**Comment by ernaneSNcarvalho** [comment]
Realmente parece ser um erro no caminho, por ser case sensitive o linux. Fiz o teste so com a instancia de fileInfo e nao apareceu erro, apos corrigir o caminho, mas quando tento fazer a copia do arquivo, atraves do copyTo(targetPath) ele aparece o seguinte erro: 

An error occurred
Could not find a part of the path '/home/ernane/Documentos/dev/Archive/Archive1/home/ernane/Teste.txt'.

e codigo: 
class Program
    {
        static void Main(string[] args)
        {
            string sourcePath = @"home/ernane/Teste.txt";
            string targetPath = @"/home/ernane/Teste2.txt";
            
            try
            {
                FileInfo fileInfo = new FileInfo(sourcePath);
                fileInfo.CopyTo(targetPath);
            }
            catch(IOException e)
            {
                Console.WriteLine("An error occurred");
                Console.WriteLine(e.Message);
            }
        }

**Reply by andrebaltieri** [reply]
Me parece outro erro de caminho:
```
/home/ernane/Documentos/dev/Archive/Archive1/home/ernane/Teste.txt
```

Aparentemente `/home/ernane/` se repete... é isto mesmo?

**Reply by ernaneSNcarvalho** [reply]
Sim. A intenção era criar uma cópia de um arquivo no mesmo diretório.

**Reply by andrebaltieri** [reply]
Sim, mas os caminhos estão incorretos imagino:
```
string sourcePath = @"home/ernane/Teste.txt";
string targetPath = @"/home/ernane/Teste2.txt";
```

**Reply by ernaneSNcarvalho** [reply]
Desculpe a insistência e ignorância quanto ao sistema operacional (estou quase voltando para o windows kk).

Vou reformular minha pergunta: 

Para acessar um arquivo em um diretório no windows: 
Ex: c:/temp/arquivo.txt (simples assim)

Para acessar um arquivo no Linux: 
????

Acho que essa resposta ai ja vai me ajudar.

**Reply by andrebaltieri** [reply]
Bom dia @ernaneSNcarvalho, no .NET a sintaxe não, o que muda são os caminhos no seu SO.
Você pode descobrir o caminho de um arquivo assim:
```csharp
var fileInfo = new FileInfo("teste.txt");
Console.WriteLine(fileInfo.FullName);
```

Aqui tem um exemplo que rodei no .NET Fiddle (Máquina Linux) e funcionou:
```csharp
using System;
using System.IO;

public class Program
{
	public static void Main()
	{
		try
		{
			using var sw = new StreamWriter("teste.txt");
			sw.Write("hello");
			
		}
		catch (IOException e)
		{
			Console.WriteLine("An error occurred");
			Console.WriteLine(e.Message);
		}

		try
		{
			var fileInfo = new FileInfo("teste.txt");
			fileInfo.CopyTo("teste2.txt");
			
			Console.WriteLine(fileInfo.FullName);
		}
		catch (IOException e)
		{
			Console.WriteLine("An error occurred");
			Console.WriteLine(e.Message);
		}
	}
}
```

**Reply by ernaneSNcarvalho** [reply]
Funcionou
